### PR TITLE
Add choice type for resource group location.

### PR DESCRIPTION
**Pull request by Smarker**
No need for location validation, now that location is limited to a set of values.

Addresses #25

### CODE REVIEW

**Comment by EricSchles** [comment]
I love that you are using hypothesis.  Yessssssss.  Okay going to review this in detail

**Comment by Smarker** [comment]
As a side note - I knew that I could have done something like:

```
subscription_client = SubscriptionClient(creds)
    locations = subscription_client.subscriptions.list_locations(subscription_id)
    for location in locations:
        print(location.name)
```

But to call `list_locations` I would need the aad parameters, and I did not want to force aad args to appear before the location parameter. This is why I just ended up providing it as a constant and not a list generated from calling `list_locations` on the current subscription.

File: cli/regex.py [hunk 1]
```diff
@@ -1,5 +1,4 @@
 RG_NAME_PATTERN = '^[-\w\._\(\)]{1,90}[^\.]$'
```
Comment (EricSchles) [inline cli/regex.py:1]: The use of regex is great!  However you should adopt the commenting style as specified here: 

https://docs.python.org/3/library/re.html#re.VERBOSE

Explaining what each piece of the regex means is super helpful with those not familiar with your pattern.

File: cli/regex.py [hunk 2]
```diff
@@ -1,5 +1,4 @@
...
-LOCATION_PATTERN = '^[a-zA-Z]+2?$'
 STORAGE_NAME_PATTERN = '^[a-z0-9]{3,24}$'
```
Comment (EricSchles) [inline cli/regex.py:2]: Same for here as comment on line 1.

**Review by EricSchles — COMMENTED** [review]
I think that's a very wise design choice!

File: cli/regex.py (hunk 2 above)
Comment (Smarker) [reply cli/regex.py:2]: I actually removed the `LOCATION_PATTERN` so I don't need to document it here, but I'll definitely document the other regexes in a separate PR.

**Review by EricSchles — COMMENTED** [review]
Code generally looks good!  happy with the available_regions object.  Very smart.

**Review by EricSchles — APPROVED** [review]

File: cli/regex.py (hunk 1 above)
Comment (Smarker) [reply cli/regex.py:1]: after speaking more with @chrobles, we want to defer this input validation to azure. it's not necessary to do the regex validation checking if azure will already do that for us. we can still put in checks in place to make sure necessary resources are created though

**Pull request merged**